spynnaker/pyNN/models/neuron/population_machine_vertex.py

```diff
@@ -467,10 +467,15 @@ def generate_data_specification(
         spec.write_array(simulation_utilities.get_simulation_header_array(
             self.__binary_file_name, machine_time_step, time_scale_factor))
 
+        max_rewires_per_ts = 1
+        s_dynamics = self._app_vertex.synapse_manager.synapse_dynamics
+        if isinstance(s_dynamics, AbstractSynapseDynamicsStructural):
+            max_rewires_per_ts = s_dynamics.get_max_rewires_per_ts(
+                machine_time_step)
```

Comment:
This might end up in the constructor where the neuron recorder gets made where you can then pass it in to the constructor.